auth_saml/controllers/main.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (C) 2020 GlodoUK <https://www.glodo.uk/>
+# Copyright (C) 2010-2016 XCG Consulting <http://odoo.consulting>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import functools
+import json
+import logging
+
+import werkzeug.utils
+from werkzeug.urls import url_quote_plus
+
+import odoo
+from odoo import SUPERUSER_ID, _, api, http, registry as registry_get
+from odoo.http import request
+
+from odoo.addons.web.controllers.main import (
+    Home,
+    ensure_db,
+    login_and_redirect,
+    set_cookie_and_redirect,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+# ----------------------------------------------------------
+# helpers
+# ----------------------------------------------------------
+
+
+def fragment_to_query_string(func):
+    @functools.wraps(func)
+    def wrapper(self, req, **kw):
+        if not kw:
+            return """<html><head><script>
+                var l = window.location;
+                var q = l.hash.substring(1);
+                var r = '/' + l.search;
+                if(q.length !== 0) {
+                    var s = l.search ? (l.search === '?' ? '' : '&') : '?';
+                    r = l.pathname + l.search + s + q;
+                }
+                window.location = r;
+            </script></head><body></body></html>"""
+        return func(self, req, **kw)
+
+    return wrapper
+
+
+# ----------------------------------------------------------
+# Controller
+# ----------------------------------------------------------
+
+
+class SAMLLogin(Home):
+    def _list_saml_providers_domain(self):
+        return [
+            ("sp_pem_public", "!=", False),
+            ("sp_pem_private", "!=", False),
+        ]
```

Comment:
I find this a bit disturbing in practice. Shouldn't we rely only on `active` boolean on the IDP config?

Otherwise we need some helper on the IDP form to explain why it's not visible on the login page.

Reply:
As we override the sp_pem_public and sp_pem_private variable with additional variable with paths, we will change the behavior in `auth_saml_environment` module. (It's also a bit tricky to inherit in a clean way this method without overwritting it as super doesn't react well)

Reply:
There was a reason for this, but it's not relevant any more. I've altered the domain to effectively be empty :) Thank you!